src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -209,9 +209,10 @@ export async function generateRawTransaction(args: {
 }): Promise<RawTransaction> {
   const { aptosConfig, sender, payload, options } = args;
 
-  const getSequenceNumber = options?.accountSequenceNumber
-    ? Promise.resolve({ sequence_number: options.accountSequenceNumber })
-    : getInfo({ aptosConfig, accountAddress: sender });
+  const getSequenceNumber =
+    options?.accountSequenceNumber !== undefined
+      ? Promise.resolve({ sequence_number: options.accountSequenceNumber })
+      : getInfo({ aptosConfig, accountAddress: sender });
```

Comment:
We will probably want to verify anywhere else we use ternary operators and other places where we're trying to take advantage of falsiness then

Reply:
Certainly, but the appropriateness of using ternary operators and leveraging falsiness depends on the specific use case.

Reply:
I'd prefer to avoid using falsiness / truthiness as much as possible to prevent mistakes like this